backends/sycl-gen/ceed-sycl-gen-operator.sycl.cpp

```diff
@@ -0,0 +1,188 @@
+//------------------------------------------------------------------------------
+static int CeedOperatorApplyAdd_Sycl_gen(CeedOperator op, CeedVector input_vec, CeedVector output_vec, CeedRequest *request) {
+  Ceed ceed;
+  CeedCallBackend(CeedOperatorGetCeed(op, &ceed));
+  Ceed_Sycl *ceed_Sycl;
+  CeedCallBackend(CeedGetData(ceed, &ceed_Sycl));
+  CeedOperator_Sycl_gen *impl;
+  CeedCallBackend(CeedOperatorGetData(op, &impl));
+  CeedQFunction           qf;
+  CeedQFunction_Sycl_gen *qf_impl;
+  CeedCallBackend(CeedOperatorGetQFunction(op, &qf));
+  CeedCallBackend(CeedQFunctionGetData(qf, &qf_impl));
+  CeedInt num_elem, num_input_fields, num_output_fields;
+  CeedCallBackend(CeedOperatorGetNumElements(op, &num_elem));
+  CeedOperatorField *op_input_fields, *op_output_fields;
+  CeedCallBackend(CeedOperatorGetFields(op, &num_input_fields, &op_input_fields, &num_output_fields, &op_output_fields));
+  CeedQFunctionField *qf_input_fields, *qf_output_fields;
+  CeedCallBackend(CeedQFunctionGetFields(qf, NULL, &qf_input_fields, NULL, &qf_output_fields));
+  CeedEvalMode eval_mode;
+  CeedVector   vec, output_vecs[CEED_FIELD_MAX] = {};
+
+  // Creation of the operator
+  CeedCallBackend(CeedSyclGenOperatorBuild(op));
+
+  // Input vectors
+  for (CeedInt i = 0; i < num_input_fields; i++) {
+    CeedCallBackend(CeedQFunctionFieldGetEvalMode(qf_input_fields[i], &eval_mode));
+    if (eval_mode == CEED_EVAL_WEIGHT) {  // Skip
+      impl->fields->inputs[i] = NULL;
+    } else {
+      // Get input vector
+      CeedCallBackend(CeedOperatorFieldGetVector(op_input_fields[i], &vec));
+      if (vec == CEED_VECTOR_ACTIVE) vec = input_vec;
+      CeedCallBackend(CeedVectorGetArrayRead(vec, CEED_MEM_DEVICE, &impl->fields->inputs[i]));
+    }
+  }
+
+  // Output vectors
+  for (CeedInt i = 0; i < num_output_fields; i++) {
+    CeedCallBackend(CeedQFunctionFieldGetEvalMode(qf_output_fields[i], &eval_mode));
+    if (eval_mode == CEED_EVAL_WEIGHT) {  // Skip
+      impl->fields->outputs[i] = NULL;
+    } else {
+      // Get output vector
+      CeedCallBackend(CeedOperatorFieldGetVector(op_output_fields[i], &vec));
+      if (vec == CEED_VECTOR_ACTIVE) vec = output_vec;
+      output_vecs[i] = vec;
+      // Check for multiple output modes
+      CeedInt index = -1;
+      for (CeedInt j = 0; j < i; j++) {
+        if (vec == output_vecs[j]) {
+          index = j;
+          break;
+        }
+      }
+      if (index == -1) {
+        CeedCallBackend(CeedVectorGetArray(vec, CEED_MEM_DEVICE, &impl->fields->outputs[i]));
+      } else {
+        impl->fields->outputs[i] = impl->fields->outputs[index];
+      }
+    }
+  }
+
+  // Get context data
+  CeedCallBackend(CeedQFunctionGetInnerContextData(qf, CEED_MEM_DEVICE, &qf_impl->d_c));
+
+  // Apply operator
+  const CeedInt dim       = impl->dim;
+  const CeedInt Q_1d      = impl->Q_1d;
+  const CeedInt P_1d      = impl->max_P_1d;
+  const CeedInt thread_1d = CeedIntMax(Q_1d, P_1d);
```

Comment:
```suggestion
```
I'm getting "unused variable" warnings for `thread_2d`

Reply:
I have removed this variable in the following patch